### PR TITLE
podman5 machine reconfig 1

### PR DESCRIPTION
**Pull request by baude**
this is a logical place to get changes upstream before they grow out of control.  this pr is the first in an effort to deduplicate machine code and streamline code flow.

a lot of code is simply moved to eliminate circular imports.  names and specific paths can ultimately be changed.  i dont like some of the descriptive interface names, etc.  ultimately, i think once we have the "old" code sanitized, we can re-use some of those.

clearly some of what is in here is temporary and will either be deleted, changed, or moved again as this effort comes to a close.

right now, the machine code does not use any of the "new" code.  you will see in `init` and `rm` some commented out code that hooks it. i'm afraid things will get worse before they get better (way worse).

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by cgwalters** [comment]
There's definitely some background I'm missing on what the higher level goal here is (would be good to have e.g. a tracking issue or so?) but...at least *some* of this seems like it could be done incrementally?

**Comment by baude** [comment]
Talked to the team ... agreed I will work in a dev branch and reviews will occur by provider.  Otherwise it is just way too hard to organize.

File: pkg/machine/vmconfigs/machine.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  set         Set a virtual machine setting				specific
+  ssh         SSH into an existing machine				common
+  start       Start an existing machine					specific
+  stop        Stop an existing machine					specific
+*/
+
+var (
+	SSHRemoteConnection     RemoteConnectionType = "ssh"
+	DefaultIgnitionUserName                      = "core"
+	ForwarderBinaryName                          = "gvproxy"
```
Comment (jakecorrenti) [inline pkg/machine/vmconfigs/machine.go:37]: `DefaultIgnitionUserName` is already defined in `pkg/machine/config.go` and in the `ignition package`. Can we find a common place? maybe define somewhere? `ForwarderBinaryName` is also already defined in `pkg/machine/config.go`

File: pkg/machine/qemuprovider.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package machine
```
Comment (jakecorrenti) [inline pkg/machine/qemuprovider.go:1]: can we split the file name into something like `pkg/machine/qemu_provider.go`

**Pull request closed without merging**